kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -379,10 +379,17 @@ def production(self) -> KProduction:
 
     @property
     def macro_bin_runtime(self) -> KRule:
+        if self.has_unlinked():
+            raise AssertionError(
+                'Some library placeholders have been found. Please link the library(ies) first. Ref: https://docs.soliditylang.org/en/v0.8.20/using-the-compiler.html#library-linking'
+            )
         return KRule(
             KRewrite(KEVM.bin_runtime(KApply(self.klabel)), KEVM.parse_bytestack(stringToken('0x' + self.bytecode)))
         )
 
+    def has_unlinked(self) -> bool:
+        return any(self.bytecode[i] == '__' for i in range(0, len(self.bytecode), 2))
+
```

Comment:
It seems a simpler solution (and probably faster):

```suggestion
    def has_unlinked(self) -> bool:
        return 0 <= self.bytecode.find('__')

```

Reply:
![image](https://github.com/runtimeverification/evm-semantics/assets/12721972/0c3d61dd-d8ff-4d13-bc60-a8bede06b275)

Reply:
Nice thanks!